### PR TITLE
Added JIS version for Keychron Q2

### PR DESCRIPTION
**Pull request by lalalademaxiya1**
As the title.

## Description

- Added q2 jis version and it's encoder variant.
- Added capslock indication for all q2 keyboards.
- Replaced MCU name in rule.mk file of ansi and iso versions with STM32L432 instead of STM32L433.
- Rewrote the light effect definition to keep up with other Keychron's keyboards.

- Removed MCU and bootloader definitions from rule.mk and replaced them in info.json file.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keychron/q2/readme.md [hunk 1]
```diff
@@ -8,11 +8,15 @@ A customizable 65% keyboard.
 
 Make example for this keyboard (after setting up your build environment):
 
-  make keychron/q2/ansi:default
+```bash
+make keychron/q2/ansi:default
+```
```
Comment (waffle87) [inline keyboards/keychron/q2/readme.md:13]:
```suggestion
    make keychron/q2/ansi:default
```

File: keyboards/keychron/q2/readme.md [hunk 2]
```diff
@@ -8,11 +8,15 @@ A customizable 65% keyboard.
...
 
 Flashing example for this keyboard ([after setting up the bootloadHID flashing environment](https://docs.qmk.fm/#/flashing_bootloadhid))
 
-  make keychron/q2/ansi:default:flash
+```bash
+make keychron/q2/ansi:default:flash
+```
```
Comment (waffle87) [inline keyboards/keychron/q2/readme.md:19]:
```suggestion
    make keychron/q2/ansi:default:flash
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q2/config.h [hunk 3]
```diff
@@ -103,4 +103,7 @@
 #define ENABLE_RGB_MATRIX_SPLASH
 // #define ENABLE_RGB_MATRIX_MULTISPLASH
 #define ENABLE_RGB_MATRIX_SOLID_SPLASH
-#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
+// #define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
+
+#define RGB_MATRIX_KEYPRESSES
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
```
Comment (adophoxia) [inline keyboards/keychron/q2/config.h:109]: This was already addressed in the previous KC that was merged earlier.
* https://github.com/qmk/qmk_firmware/pull/19685/

File: keyboards/keychron/q2/jis_encoder/info.json [hunk 4]
```diff
@@ -0,0 +1,92 @@
+{
+    "keyboard_name": "Keychron Q2",
+    "manufacturer": "Keychron",
+    "url": "https://github.com/Keychron",
+    "maintainer": "lalalademaxiya1",
+    "usb": {
+        "vid": "0x3434",
+        "pid": "0x0115",
+        "device_version": "1.0.0"
+    },
```
Comment (fauxpark) [inline keyboards/keychron/q2/jis_encoder/info.json:10]:
```suggestion
    },
    "processor": "STM32L432",
    "bootloader": "stm32-dfu",
```

File: keyboards/keychron/q2/jis_encoder/rules.mk [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# MCU name
+MCU = STM32L432
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
```
Comment (fauxpark) [inline keyboards/keychron/q2/jis_encoder/rules.mk:6]:
```suggestion
```

File: keyboards/keychron/q2/jis/info.json [hunk 6]
```diff
@@ -0,0 +1,92 @@
+{
+    "keyboard_name": "Keychron Q2",
+    "manufacturer": "Keychron",
+    "url": "https://github.com/Keychron",
+    "maintainer": "lalalademaxiya1",
+    "usb": {
+        "vid": "0x3434",
+        "pid": "0x0114",
+        "device_version": "1.0.0"
+    },
```
Comment (fauxpark) [inline keyboards/keychron/q2/jis/info.json:10]:
```suggestion
    },
    "processor": "STM32L432",
    "bootloader": "stm32-dfu",
```

File: keyboards/keychron/q2/jis/rules.mk [hunk 7]
```diff
@@ -0,0 +1,27 @@
+# MCU name
+MCU = STM32L432
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
```
Comment (fauxpark) [inline keyboards/keychron/q2/jis/rules.mk:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
And same for ANSI & ISO.

**Comment by waffle87** [comment]
> And same for ANSI & ISO.

The same changes need to be applied for the ANSI & ISO versions (remove MCU & bootloader definitions from `rules.mk`, replace them in `info.json` file)

File: keyboards/keychron/q2/config.h (hunk 3 above)
Comment (lalalademaxiya1) [reply keyboards/keychron/q2/config.h]: All right, correct now, thanks~

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**